ckanext/external_storage/cli.py

```diff
@@ -0,0 +1,266 @@
+import errno
+import logging
+import os
+import shutil
+import tempfile
+from contextlib import contextmanager
+from typing import Any, Dict, Generator, Tuple
+
+import requests
+from ckan.lib.cli import CkanCommand
+from ckan.lib.helpers import _get_auto_flask_context  # noqa  we need this for Flask request context
+from ckan.model import Resource, Session, User
+from ckan.plugins import toolkit
+from flask import Response
+from giftless_client import LfsClient
+from giftless_client.types import ObjectAttributes
+from six import binary_type, string_types
+from sqlalchemy import or_
+from sqlalchemy.orm.attributes import flag_modified
+from werkzeug.wsgi import FileWrapper
+
+from ckanext.external_storage import helpers
+from ckanext.external_storage.download_handler import call_download_handlers
+
+
+def _log():
+    return logging.getLogger(__name__)
+
+
+class MigrateResourcesCommand(CkanCommand):
+    """Migrate all non-migrated resources to external blob storage
+    """
+    summary = __doc__.split('\n')[0]
+    usage = __doc__
+    min_args = 0
+
+    _user = None
+    _max_failures = 5
+
+    def command(self):
+        self._load_config()
+        self._user = User.get(self.site_user['name'])
+        with app_context() as context:
+            context.g.user = self.site_user['name']
```

Comment:
This should not be necessary because actions, auth etc should not rely on a user existing in `g` (ie being in the context of a web request) but I know it sometimes happens in CKAN so if you found issues with this your approach here is correct.
I think the place where `g.user` is needed is [this one](https://github.com/datopian/ckanext-authz-service/blob/master/ckanext/authz_service/authz_binding/common.py#L62). I don't know the implications on ckanext-authz-service but if you wanted to support this scenario at that point you can check if `g` has a `user` attribute. If it does it's a web request (if `g.user` is None is a non-authenticated (non-logged in) request), if it doesn't it's a test / script / cli call and you can choose what to do (fallback to the site user, no user, etc)
But obviously you'll know better if it's worth doing.

`app_context()` is needed because you are using `send_file()` down the line so that's good

Reply:
Thanks, you are correct about why I needed `g.user`. It may be something worth fixing in `ckanext-authz-service` and not here. I'll look into it. 

Also correct about `app_context()`. Using the private `_get_auto_flask_context` from `ckan.lib.helpers` was the only reasonable way I found to doing that without duplicating tons of code. I'm worried about using a private function, so I was wondering if you have a better suggestion.

Reply:
I looked at the `ckanext-authz-service` code and while I can make it more resiliant to not having `g.user` set, I'm not sure what the effects of that will be in terms of the authorization tokens I get. For now I'm going to merge this, and will make these changes in ckanext-authz-service anyway.